abci/types/application.go

```diff
@@ -98,11 +101,17 @@ func (BaseApplication) ApplySnapshotChunk(req RequestApplySnapshotChunk) Respons
 }
 
 func (BaseApplication) PrepareProposal(req RequestPrepareProposal) ResponsePrepareProposal {
-	return ResponsePrepareProposal{BlockData: req.BlockData}
+	squareSizeBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(squareSizeBytes, 0)
+
+	// placeholderDataHash and placeholderSquareSizeBytes are expected to be
+	// overwritten by celestia-app.
```

Comment:
[blocking] it's a bit confusing for this comment to exist b/c it references variables that don't exist.